src/components/ButtonWithDropdownMenu.js

```diff
@@ -54,16 +54,26 @@ const ButtonWithDropdownMenu = (props) => {
     const [popoverAnchorPosition, setPopoverAnchorPosition] = useState(null);
     const {windowWidth, windowHeight} = useWindowDimensions();
     const caretButton = useRef(null);
-    useEffect(() => {
-        if (!caretButton.current) {
-            return;
-        }
+
+    const measurePopoverPosition = () => {
         caretButton.current.measureInWindow((x, y, w, h) => {
             setPopoverAnchorPosition({
                 horizontal: x + w,
                 vertical: y + h,
             });
+            setIsMenuVisible(true);
         });
+    };
```

Comment:
Rename this function to something more fitting to its functionality e.g. `measurePopoverPositionAndShowMenu` or anything that may be better.

Reply:
@s77rt changed function name into measurePopoverPositionAndShowMenu